src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -1416,7 +1417,7 @@ class ApplicationManagerImpl
   // Thread that pumps messages audio pass thru to mobile.
   impl::AudioPassThruQueue audio_pass_thru_messages_;
 
-  HMICapabilities hmi_capabilities_;
+  std::auto_ptr<HMICapabilities> hmi_capabilities_;
```

Comment:
@LevchenkoS I have to say that in the C++11 standard auto_ptr is declared deprecated. So future use of this feature may cause problems. I suggested SharedPtr as it is a wrapper which can effectively handle problems like this. And if there was a functionality like C++11 unique_ptr I would have suggested you to use it.